backend/src/routes/api/modelRegistries/index.ts

```diff
@@ -35,29 +37,22 @@ export default async (fastify: KubeFastifyInstance): Promise<void> => {
     ),
   );
 
+  // Accepts a ModelRegistry and a password, creates the MR and an associated Secret
+  // Returns the ModelRegistry only
   fastify.post(
     '/',
     secureAdminRoute(fastify)(
       async (
         request: FastifyRequest<{
           Querystring: { dryRun?: string };
-          Body: ModelRegistryKind;
+          Body: ModelRegistryAndDBPassword;
         }>,
         reply: FastifyReply,
       ) => {
         const { dryRun } = request.query;
-        const modelRegistry = request.body;
+        const { modelRegistry, databasePassword } = request.body;
         try {
-          const response = await fastify.kube.customObjectsApi.createNamespacedCustomObject(
-            'modelregistry.opendatahub.io',
-            'v1alpha1',
-            MODEL_REGISTRY_NAMESPACE,
-            'modelregistries',
-            request.body,
-            undefined,
-            dryRun,
-          );
-          return response.body;
+          return createModelRegistryAndSecret(fastify, modelRegistry, databasePassword, !!dryRun);
```

Comment:
if `dryRun` has a "string" type, could it be "false"? If so, should this be changed to account for that?

Reply:
In the k8s API, `dryRun` can only either be undefined or `"All"`. It's a string in case future API versions add more types of dry-runs I guess.

https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.24/
> When present, indicates that modifications should not be persisted. An invalid or unrecognized dryRun directive will result in an error response and no further processing of the request. Valid values are: - All: all dry run stages will be processed

We could maybe use a more specific type `"All" | undefined` on our end, but the fastify methods we're calling expect a plain `string`.